packages/block-library/src/cover/style.scss

```diff
@@ -96,15 +95,14 @@
 
 	.wp-block-cover__inner-container {
 		width: 100%;
-		z-index: z-index(".wp-block-cover__inner-container");
 		color: inherit;
 		// Reset the fixed LTR direction at the root of the block in RTL languages.
 		/*rtl:raw: direction: rtl; */
+	}
 
-		// Reset the z-index to auto when the body has a modal open. So when the
-		// modal is inside the cover, it doesn't create a stacking context.
-		@at-root .has-modal-open & {
-			z-index: auto;
+	&.has-no-zindex-in-inner-blocks {
+		.wp-block-cover__inner-container {
+			position: relative; // It needs a position to be on top of the other elements.
```

Comment:
Please, look at this property with a grain of salt. I think it won't cause any issue, but if we had an element inside this block with a position absolute, it would be affected.

The next block in the tree with the position relative would be the immediate parent (`.wp-block-cover`). I don't believe we would have an existing think trusting in the position relative of the `.wp-block-cover`, but we never know! =P

Reply:
After a quick glance it seems okay. Once the deprecation and styles are iterated on we can test this aspect after they've settled.